aiojobs/aiohttp.py

```diff
@@ -60,10 +61,17 @@ async def wrapper(request_or_view: _RequestView) -> _T:
     return wrapper
 
 
-def setup(app: web.Application, **kwargs: Any) -> None:
+def setup(
+    app: web.Application, graceful_timeout: Optional[float] = None, **kwargs: Any
+) -> None:
     async def cleanup_context(app: web.Application) -> AsyncIterator[None]:
         app["AIOJOBS_SCHEDULER"] = scheduler = Scheduler(**kwargs)
         yield
+        if graceful_timeout and scheduler._jobs:
+            await asyncio.gather(
+                *[job.wait(timeout=graceful_timeout) for job in scheduler._jobs],
+                return_exceptions=True,
+            )
```

Comment:
Ideally, we want this logic to be in a `scheduler.wait_and_close()` method, then it can be reused for other applications as well.

Otherwise, looks reasonable. Will come back for a full review in a few days.